x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -130,27 +134,58 @@ public void query(List<Attribute> output, QueryContainer query, String index, Ac
         if (query.isAggsOnly()) {
             if (query.aggs().useImplicitGroupBy()) {
                 l = new ImplicitGroupActionListener(listener, client, cfg, output, query, search);
+                // execute the request right away, no need for a PIT
+                client.search(search, l);
+                return;
             } else {
                 l = new CompositeActionListener(listener, client, cfg, output, query, search);
             }
         } else {
-            search.scroll(keepAlive);
-            l = new ScrollActionListener(listener, client, cfg, output, query);
+            l = new SearchHitActionListener(listener, client, cfg, output, query, search);
+        }
+
+        openPointInTime(index, query.shouldIncludeFrozen(), search, l);
+    }
+
+    private void openPointInTime(String index, boolean includeFrozen, SearchRequest search, ActionListener<SearchResponse> listener) {
+        final OpenPointInTimeRequest openPitRequest = new OpenPointInTimeRequest(Strings.commaDelimitedListToStringArray(index),
+            indicesOptions(includeFrozen), keepAlive, null, null);
+        client.execute(OpenPointInTimeAction.INSTANCE, openPitRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(OpenPointInTimeResponse openPointInTimeResponse) {
+                String searchContextId = openPointInTimeResponse.getSearchContextId();
+                search.source().pointInTimeBuilder(new SearchSourceBuilder.PointInTimeBuilder(searchContextId, null));
+                client.search(search, listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    public static void closePointInTime(Client client, String pointInTimeId, ActionListener<Boolean> listener) {
+        if (pointInTimeId != null) {
+            client.execute(ClosePointInTimeAction.INSTANCE, new ClosePointInTimeRequest(pointInTimeId),
+                wrap(clearPointInTimeResponse -> listener.onResponse(clearPointInTimeResponse.isSucceeded()), listener::onFailure));
+        } else {
+            listener.onResponse(false);
         }
+    }
 
-        client.search(search, l);
+    private static IndicesOptions indicesOptions(boolean includeFrozen) {
+        return includeFrozen ? IndexResolver.FIELD_CAPS_FROZEN_INDICES_OPTIONS : IndexResolver.FIELD_CAPS_INDICES_OPTIONS;
     }
 
-    public static SearchRequest prepareRequest(Client client, SearchSourceBuilder source, TimeValue timeout, boolean includeFrozen, 
+    public static SearchRequest prepareRequest(Client client, SearchSourceBuilder source, TimeValue timeout, boolean includeFrozen,
             String... indices) {
         return client.prepareSearch(indices)
-                // always track total hits accurately
-                .setTrackTotalHits(true).setAllowPartialSearchResults(false).setSource(source).setTimeout(timeout)
```

Comment:
`setTrackTotalHits(true)` call is effectively discarded by subsequent `setSource(source)`.
While this could be fixed in another PR, it being ignored seems harmless (and thus #52787 could be closed by this issue as well).